### ISSUE TITLE
minor ticks aren't inheriting pars

### ISSUE THREAD

**Issue by ldecicco-USGS** [USGS-R/gsplot#437]
```
gsplot(frame.plot=FALSE) %>%
   background_color(col="grey90") %>%
   grid(col = "white", lty = 1) %>%
   par(tcl=-0.2, cex=0.75, xaxs= "r", yaxs =  "r") %>%
   axis(side = c(1,2), lwd = 0, lwd.ticks = 1, n.minor=1)
```
Should have the minor tick marks going outside, shows up inside:

![image](https://cloud.githubusercontent.com/assets/1105215/22940508/b4bee3a8-f2a7-11e6-8194-a9c2f442e57f.png)

**Comment by ldecicco-USGS** [comment]
Also:
```
gs <- gsplot() %>%
  points(1:100, 1:100, log="xy") %>%
  axis(side=c(3,4), labels=FALSE, n.minor=4) %>%
  axis(side=c(1,2), n.minor=4)
gs

```
side's 3 and 4 ticks aren't logged (and should be).

**Comment by lindsayplatt** [comment]
:point_up: @ldecicco-USGS I'm assuming this is example is something to do with #436 ?

**Comment by ldecicco-USGS** [comment]
They might be related, but that issue is specific to the date_axis.

**Comment by ldecicco-USGS** [comment]
Actually, that second example I posted has nothing to do with minor ticks. This:

```
gsplot() %>%
   points(1:100, 1:100, log="xy") %>%
   axis(side=c(3,4)) 
```

Shows a linear axis on 3 and 4 (and log on 1, 2). I would expect 3 and 4 to be logged if no other call to 3 & 4 are defined.

**Comment by lindsayplatt** [comment]
Yeah, that second example seems like a second issue.

**Comment by ldecicco-USGS** [comment]
Yup, making a new issue.

**Comment by lindsayplatt** [comment]
I'm seeing logged ticks now, but still seeing minor ticks on the inside instead of outside:

```r
gsplot(frame.plot=FALSE) %>%
   background_color(col="grey90") %>%
   grid(col = "white", lty = 1) %>%
   par(tcl=-0.2, cex=0.75, xaxs= "r", yaxs =  "r") %>%
   axis(side = c(1,2), lwd = 0, lwd.ticks = 1, n.minor=1)
```
![image](https://cloud.githubusercontent.com/assets/13220910/24115142/71e10f78-0d70-11e7-8241-b13e6978148f.png)


```r
gs <- gsplot() %>%
    points(1:100, 1:100, log="xy") %>%
    axis(side=c(3,4), n.minor=4) %>%
    axis(side=c(1,2), n.minor=4)
gs
```

![image](https://cloud.githubusercontent.com/assets/13220910/24115126/6413665c-0d70-11e7-830d-a4e8bbcac888.png)

**Comment by lindsayplatt** [comment]
#436 seems to duplicate the remaining bug here. @ldecicco-USGS @jiwalker-usgs please verify this.

**Comment by lindsayplatt** [comment]
Wait, I did this earlier in this thread.... @ldecicco-USGS concluded that they were not the same. Sorry.

**Comment by lindsayplatt** [comment]
@ldecicco-USGS we have an arg called `tcl.minor`. That fixes the issue, but @jiwalker-usgs is thinking we should have the default match the `tcl` and `tcl.minor` needs to be used if you want them to be different. Either way, we need more documentation about this.

```r
gsplot() %>%
    grid(col = "white", lty = 1) %>%
    par(tcl=-0.2, cex=0.75, xaxs= "r", yaxs =  "r") %>%
    axis(side = c(1,2), lwd = 0, lwd.ticks = 1, n.minor=1, tcl.minor=-0.2)
```
![image](https://cloud.githubusercontent.com/assets/13220910/24115795/86ba7400-0d72-11e7-9456-5ad1978302d4.png)

**Comment by lindsayplatt** [comment]
Default `tcl.minor = par('tcl')*0.5` so that it gets the correct direction. Add documentation about tcl.minor.

**Comment by jordansread** [comment]
Thought this would be relatively simple. issues I see here:
1) `tcl` gets ignored in `axis` already, which is 👎 
```r
points(gsplot(), 1, 0, tcl = -0.5) %>% axis(side=1)
#differs from 
points(gsplot(), 1, 0) %>% axis(side=1, tcl = 0.5)
```
2) axis is a call for multiple sides, so getting a default from a side par would need to be side-specific
3) There is a hierarch of getting the default for `tcl.minor` that goes like this (I may be missing one?):
  - from the input of `tcl` on the axis call (but note this one is broken as is)
  - from the side par value of `tcl` if it exists
  - from the global par value of `tcl` in the gsplot object if it exists
  - lastly, from the `par('tcl')` value if none of the others are present.

**Comment by jordansread** [comment]
closed with #460
